### PR TITLE
ARC-1519 - Improve Jira Routing for connect/app creation

### PR DESCRIPTION
**Pull request by krazziekay**
**What's in this PR?**
- The existing routes inside `routes/jira` were updated [Following: [Routes](https://hello.atlassian.net/wiki/spaces/PF/pages/1822399418/Routes+for+GH4J)]
- Test cases added for few routes. 
- Added a temporary solution for `qsh`error when adding decoded URIs in the URI path

**Why**
- All the JIRA routes were not following REST practices and were getting very messy.

**How has this been tested?**  
- Locally by testing the urls in the general pages in the connect app-descriptor(`jira-atlassian-connect-get`)
- With few test cases

**Whats Next?**
- All the JIRA routes were updated and some were added too, however only a few test cases were added. So, need to add test cases for the remaining routes.
- This is gonna cause conflicts in a lot of other tickets, so have to resolve those conflicts too.

### CODE REVIEW

File: src/jira/util/jwt.ts [hunk 1]
```diff
@@ -81,7 +81,16 @@ function decodeAsymmetricToken(token: string, publicKey: string, noVerify: boole
 
 function verifyQsh(qsh: string, req: Request): boolean {
```
Comment (gxueatlassian) [inline src/jira/util/jwt.ts:82]: Can we have some test for this bug?

Comment (krazziekay) [reply src/jira/util/jwt.ts:82]: Test case for this case has been added in `jwt.test.ts`

File: src/routes/jira/atlassian-connect/jira-atlassian-connect-get.ts [hunk 2]
```diff
@@ -137,11 +146,20 @@ const modules = {
...
 			location: "admin_plugins_menu/gh-addon-admin-section"
+		},
+		{
+			url: "/jira/configuration",
+			conditions: adminCondition,
+			name: {
+				value: "GitHub for Jira"
+			},
+			key: "gh-addon-admin-old",
+			location: "none"
```
Comment (mboudreau) [inline src/routes/jira/atlassian-connect/jira-atlassian-connect-get.ts:157]: Have we confirmed in staging that the old route still works?

File: src/routes/jira/connect/enterprise/app/jira-connect-enterprise-app-create-or-edit.ts [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
 	req: Request,
 	res: Response,
 	next: NextFunction
 ): Promise<void> => {
 	try {
 		req.log.debug("Received Jira edit app page request");
 
-		const gitHubServerAppId = req.params.id as string;
-		const app = await GitHubServerApp.getForGitHubServerAppId(parseInt(gitHubServerAppId));
+		const uuid = req.params.uuid as string;
```
Comment (mboudreau) [inline src/routes/jira/connect/enterprise/app/jira-connect-enterprise-app-create-or-edit.ts:12]: pretty sure all params are of string type?

File: src/routes/jira/connect/enterprise/app/jira-connect-enterprise-app-create-or-edit.ts [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
+		const app = uuid ? await GitHubServerApp.findForUuid(uuid) : null;
```
Comment (mboudreau) [inline src/routes/jira/connect/enterprise/app/jira-connect-enterprise-app-create-or-edit.ts:13]: `const app = await GitHubServerApp.findForUuid(uuid)` should do.  `findForUuid` should look to see if `uuid` is defined, if not should return null

Comment (mboudreau) [reply src/routes/jira/connect/enterprise/app/jira-connect-enterprise-app-create-or-edit.ts:13]: yeah, `findForUuid` doesn't do that check.  Maybe add it? just a simple `if(!uuid) { return null; }` will do

**Comment by krazziekay** [comment]
Closing this PR because its too large to be reviewed.
Breaking this PR into smaller PRs with:
- ARC-1519-1
- ARC-1519-2
- ARC-1519-3
- ARC-1519-4
- ARC-1519-5
- ARC-1519-6
- ARC-1519-7
- ARC-1519-8
- ARC-1519-9

**Pull request closed without merging**